src/vs/workbench/contrib/chat/browser/actions/chatContextActions.ts

```diff
@@ -635,17 +635,32 @@ export class AttachContextAction extends Action2 {
 		const defer = new DeferredPromise<boolean>();
 		const addPromises: Promise<void>[] = [];
 
-		store.add(qp.onDidAccept(e => {
+		store.add(qp.onDidAccept(async e => {
 			const [selected] = qp.selectedItems;
 			if (isChatContextPickerPickItem(selected)) {
-				const attachment = selected.asAttachment();
-				if (isThenable(attachment)) {
-					addPromises.push(attachment.then(v => widget.attachmentModel.addContext(v)));
+				qp.busy = true;
+				const isValidForAttachment = await selected.validateForAttachment?.().finally(() => {
```

Comment:
If we get rid of `validateForAttachment`, we would essentially selected.asAttachment here instead (keeping qp.busy=true while that's resolving).

We still want to keep `addPromises` in case the user closes the quickpick after selecting something and before `asAttachment` resolves, but if the picker is closed by that point we would do nothing if it returns `noop`